lib/srv/reexec.go

```diff
@@ -608,6 +610,105 @@ func runPark() (errw io.Writer, code int, err error) {
 	}
 }
 
+const (
+	waitNoResolveDefaultPeriod  = 10 * time.Second
+	waitNoResolveDefaultTimeout = 10 * time.Minute
+)
+
+func waitNoResolve() (io.Writer, int, error) {
+	// We get parameters from environment variables
+	utils.InitLogger(utils.LoggingForCLI, log.DebugLevel)
+
+	domainName := os.Getenv(teleport.WaitNoResolveDomainEnvVar)
+	if domainName == "" {
+		return io.Discard, teleport.DomainStillResolving,
+			fmt.Errorf("environment variable %s must be set", teleport.WaitNoResolveDomainEnvVar)
+	}
+
+	var err error
+	period := waitNoResolveDefaultPeriod
+	periodEnv := os.Getenv(teleport.WaitNoResolvePeriodEnvVar)
+	if periodEnv != "" {
+		period, err = time.ParseDuration(periodEnv)
+	}
+
+	if err != nil {
+		return io.Discard, teleport.DomainStillResolving, err
```

Comment:
Nit: Wrap errors everywhere they are returned.
```suggestion
		return io.Discard, teleport.DomainStillResolving, trace.Wrap(err)
```